Issue 200 | Software wallet integration for transaction legibility

nconsigny opened the pull request:
#200 

refactored the criterion for transaction legibility 

Update explanations and URLs for supported wallets, ensuring clarity on integration capabilities. Refactor hardware wallet support evaluation logic for better handling of unrated statuses.

nconsigny committed:
Issue 200 | Enhance hardware wallet support documentation for Ambire, Safe, and Zerion. Update explanations and URLs for supported wallets, ensuring clarity on integration capabilities. Refactor hardware wallet support evaluation logic for better handling of unrated statuses.

polymutex commented on data/software-wallets/zerion.ts:
```diff
@@ -88,12 +88,39 @@ export const zerion: SoftwareWallet = {
+                'Trezor can be used with Zerion via WalletConnect and external-wallet routes, with one extra hop.',
+              url: 'https://www.ledger.com/zerion',
+            },
+          }),
```
These explanations all mention the mechanism used to connect the hardware wallet to the software wallet. How about making that an enum map of supported connection types, instead of freeform text?

Something like this in `src/schema/features/security/hardware-wallet-support.ts`:

```typescript
/**
 * Types of hardware wallets that can be supported
 */
export enum HardwareWalletType {
  // ...
}

/**
 * Connection method between software and hardware wallet.
 */
export enum HardwareWalletConnection {
  USB = 'USB',
  WALLET_CONNECT = 'WALLET_CONNECT',
}

/**
 * Set of connection types that are supported for a single hardware wallet.
 */
export type SingleHardwareWalletSupport = NonEmptyRecord<HardwareWalletConnection, Support>;

/**
 * A record of hardware wallet types and their support status
 */
export type HardwareWalletSupport = WithRef<{
  /**
   * Record of hardware wallet types and their support status
   */
  supportedWallets: Partial<Record<HardwareWalletType, Support<SingleHardwareWalletSupport>>>;
}>;
```

This would allow the attribute evaluation logic to understand the distinction between WalletConnect-only hardware wallet support vs those that are integrated directly.

nconsigny replied:
Good idea ! will do

polymutex commented on data/software-wallets/zerion.ts:
```diff
@@ -88,12 +88,39 @@ export const zerion: SoftwareWallet = {
+          [HardwareWalletType.TREZOR]: supported({
+            ref: {
+              explanation:
+                'Trezor can be used with Zerion via WalletConnect and external-wallet routes, with one extra hop.',
```
I'm not sure we should treat hardware wallets that only work over WalletConnect as "supported", because the use of WalletConnect implies that the user needs to rely on _another_ software wallet (in addition to the one being rated here) that actually integrates the hardware wallet, in order to use that wallet in the wallet described here.

(As an aside, the use of WalletConnect in general is probably not something we want to encourage long term too, as it creates a centralizing force where a single entity (walletconnect) is able to censor transactions and gather orderflow before others.)

nconsigny replied:
Absolutely, I wanted  to keep the bar low for hw support and the bar high for hw integration. But  objectively supporting through walletconnect should be equivalent to no support. We should prioritize direct webHID / webUSB connectors

nconsigny committed:
Add enum for HardwareWalletConnection

Enhance hardware wallet support by introducing new connection types (webUSB, webHID, bluetooth) and updating wallet integration for Ambire, Coinbase, Frame, MetaMask, Phantom, Rabby, Rainbow, Safe, and Zerion. Add contributor nconsigny to relevant wallets.

nconsigny committed:
refactor hww support for correct connector assessment

nconsigny committed:
fix safe

polymutex commented on src/schema/attributes/security/software-hw-integration.ts:
```diff
@@ -276,34 +299,10 @@ export const softwareHWIntegration: Attribute<SoftwareHWIntegrationValue> = {
-    }
-
-    // Return result with references if any
+    // If wallet supports any hardware wallet, rate as unrated/grey for now
```
Planning to implement this in a future PR I assume?
